### PR TITLE
Add a doc on making PRs easier to review

### PR DESCRIPTION
**Pull request by thockin**
Too wordy?  Open to ideas on making it more succinct.  The idea was to have a handy doc we could point people at when they violate these rules.

### CODE REVIEW

File: docs/devel/faster_reviews.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+## 7. Fix feedback in a new commit
+
+Your reviewer has finally sent you some feedback on FeatureX.  You make a bunch
+of changes and ... what?  You could patch those into your commits with git
+"squash" or "fixup" logic.  But that makes your changes hard to verify.  Unless
+your whole PR is pretty trivial, you should instead put your fixups into a new
+commit and re-push.  Your reviewer can then look at that commit on its own - so
+much faster to review than starting over.
+
```
Comment (justinsb) [inline docs/devel/faster_reviews.md]: Agree with this, but I worry that we're going to end up with a git history that is a mess unless we squash these before merge.  Github makes squashing much harder than just hitting the merge button, sadly.   (Though it's probably possible to write a small script to do this and merge PRs from the command line).

Comment (thockin) [inline docs/devel/faster_reviews.md]: I'll add a note - we should squash these things at the end, if possible.  But as long as GitHubs tools suck so badly, this is the only sane way

**Comment by davidopp** [comment]
LGTM

This is awesome!

I wonder if you should add a section on what to do if you've followed all of the guidelines but your PR is still not being reviewed in a timely fashion. The tips I would put there are
* make sure the oncall assigned it a reviewer/assignee
* ping the assignee by email if you think you have waited an unreasonable amount of time; we should consider publishing a list of committers' email addresses (work or private, but something that rises above the Github firehose and gets that magic >> gmail marker on it) 
* give some guidance on how to handle the following common situation: various people have chimed in with comments, you've responded to them and think everything is addressed, and you sit and sit and nothing happens. solution here: send an explicit PTAL on the thread, and then if you still don't hear anything after a while, escalate to previous bullet

**Comment by thockin** [comment]
Good point.  Section added.  PTAL.

**Comment by wojtek-t** [comment]
Awesome doc Tim!

**Comment by davidopp** [comment]
Looks good except I wasn't sure about the "fork a new PR" thing. If you do this and then commit, will the comment history on the original PR be lost when you submit?

**Comment by derekwaynecarr** [comment]
This is a good doc.

**Comment by thockin** [comment]
Yes, if you fork a PR you will lose history, but I think it is a viable
option for some situations.

**Comment by davidopp** [comment]
LGTM

**Comment by alex-mohr** [comment]
Seems enough consensus -- merging.
